FirebaseRemoteConfigSwift/Sources/PropertyWrapper/RemoteConfigValueObservable.swift

```diff
@@ -27,20 +27,23 @@ internal class RemoteConfigValueObservable<T: Decodable>: ObservableObject {
   init(key: String, remoteConfig: RemoteConfig) {
     self.key = key
     self.remoteConfig = remoteConfig
-    configValue = try! remoteConfig.configValue(forKey: key).decoded()
-    registration = remoteConfig.add(onConfigUpdateListener: { error in
+    self.configValue = try! remoteConfig.configValue(forKey: key).decoded(asType: T.self)
```

Comment:
How about?
```
do {
 self.configValue =  try remoteConfig.configValue(forKey: key).decoded(asType: T.self)
} catch {
 fatalError("RemoteConfig Internal Error: Failed to decode value for key:\(key) with error \(error)")
}
```

Reply:
I don't think we should raise a fatal error because there is no way for the developer to handle it. Property wrappers don't have a good mechanism for handling errors so we'll probably have to follow what was done with the Firestore property wrapper.
1. Wrap the property in a result type
2. Make the property an optional
3. Extend the API to take a default value to fall back to

Firestore's PW supports 1 & 2. I think supporting all three could be good here.

Reply:
Good point! I was addressing the immediate issue of improving the `!` crash potential, but your suggestion is much better.

Reply:
Changed to optional in #10128.  More thought about error handling is probably needed.

Reply:
For 3, it's an API decision we will talk about in the API review.

For 1 and 2, @ncooke3 i'm not sure what you meant, can you give an example for each? Keep in mind this is a dynamic property wrapper that's binding with a UI component, so we can't change the way it's defined (The configValue proeprty)

As for now @paulb777 should I keep what I have and update my PR with https://github.com/firebase/firebase-ios-sdk/pull/10128?

Reply:
Also by make the property optional, @paulb777  would https://github.com/firebase/firebase-ios-sdk/pull/10128 make  developers have to pass in a default value?  As we discussed, it would be impossible for the current implementation to pass in a default value given the generic type is T.

Reply:
@chliangGoogle, here are some examples of 1 & 2. They both shift the handling of the success (non-nil) and failure (nil) states to the API consumer.

1. `@RemoteConfigProperty(key: "award_message") var awardMessage: Result<String, _>`
The type of the above property is instead a result type so the required type of the property wrapper is `Result<T, Error> where T: Decodable`.
3. `@RemoteConfigProperty(key: "award_message") var awardMessage: String?`
The type of the above property has a `?` so the required type of the property wrapper is `T? where T: Decodable`.